plan/logical_plan_test.go

```diff
@@ -857,7 +858,7 @@ func (s *testPlanSuite) TestAggPushDown(c *C) {
 		p := builder.build(stmt)
 		c.Assert(builder.err, IsNil)
 		lp := p.(LogicalPlan)
-		p, err = logicalOptimize(flagBuildKeyInfo|flagPredicatePushDown|flagPrunColumns|flagAggregationOptimize, lp.(LogicalPlan), builder.ctx, builder.allocator)
+		lp, err = logicalOptimize(flagBuildKeyInfo|flagPredicatePushDown|flagPrunColumns|flagAggregationOptimize, lp.(LogicalPlan), builder.ctx, builder.allocator)
```

Comment:
the return value `logic LogicalPlan` of `logicalOptimize` seems useless,
can it be removed and just return `error`.

Reply:
The optimized logic plan may not be the original one, so we need to return it.